### PR TITLE
FAB IRMA Integration

### PR DESCRIPTION
**Pull request by ehsan-fj**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Contributing
+We love your input! We want to make contributing to this project as easy and transparent as possible, whether it's:
+
+- Reporting a bug
+- Discussing the current state of the code
+- Submitting a fix
+- Submitting a fix
```
Comment (rafaelbernard) [inline CONTRIBUTING.md]:
```suggestion
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
+FROM python:3.8-slim-buster AS prepare
```
Comment (rafaelbernard) [inline Dockerfile]: Can we use bullseye already?
```suggestion
FROM python:3.8-slim-bullseye AS prepare
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,185 @@
...
+# IRMA FAB Integration
+This repository contains FAB relying-party APIs also a CLI tool in order to work with FAB and IRMA.
+
+## FAB Relying-Party API
+The relying-party’s API is what the FAB will call when passing the credentials the end user has requested to share.
+The credentials will be shared in the W3C verifiable credential format. This specification can be found
+[here](https://www.w3.org/TR/vc-data-model/).
+
+### How to run it on your local
+First, make sure to create `.env` file from `.env.sample` and set the value for `SINGLE_SOURCE_API_KEY` environment
```
Comment (rafaelbernard) [inline README.md]: Maybe a typo? Single source is not the case, right?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+WORKDIR /app
+
+COPY requirements.txt /app/
+
+# Dependencies
+RUN pip install -r requirements.txt
+
+COPY runserver.py /app/
+COPY scripts/ /scripts/
+COPY fab/ /app/irma_fab
```
Comment (rafaelbernard) [inline Dockerfile]:
```suggestion
COPY fab/ /app/fab
```

File: fab_irma.py [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+import qrcode
+import requests
+import subprocess
+
+RELYING_PARTY_ID = ""
+RELYING_PARTY_NAME = ""
+RELYING_PARTY_LOGO = ""
+API_KEY = ""
+IRMA_TOKEN = "secret-fake-token"
+IRMA_SERVER = "https://pnonvsmdy9.execute-api.ap-southeast-2.amazonaws.com/dev"
```
Comment (rafaelbernard) [inline fab_irma.py]: Would not be better to get IRMA_SERVER_URL from env file and not expose ours?

File: fab_irma.py [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+        LOGGER.warning("Did not receive the disclosed VC! Please try again.")
+        exit(1)
+
+    LOGGER.debug("Disclosed VC from FAB: %s", response.json())
+    _extract_attributes(response)
+
+
+def _fetch_fab_disclosed_attributes(session_id):
+    response = requests.get(
+        url=f"https://f9emnttxd6.execute-api.ap-southeast-2.amazonaws.com/demo/fab/vc/{session_id}",
```
Comment (rafaelbernard) [inline fab_irma.py]: Same. Would not be better to have this URL obscured from the public?

File: fab_irma.py (hunk 5 above)
Comment (ehsan-fj) [reply fab_irma.py]: Wanted to remove but forgot! thanks for reminding

File: README.md (hunk 3 above)
Comment (rafaelbernard) [reply README.md]: Not solved yet.

Comment (ehsan-fj) [reply README.md]: done

**Review by rafaelbernard — APPROVED** [review]
Good job!

**Pull request merged**